src/sentry/api/endpoints/organization_events_trace.py

```diff
@@ -134,49 +156,104 @@ def get(self, request, organization, trace_id):
-        """We don't get the error map for the light view
+    def get_current_transaction(self, transactions, errors, event_id):
+        """Given an event_id return the related transaction event
 
-        This is because we only get spans for the root + current event, which means we could only create an error
-        to transaction association for up to two events.
+        The event_id could be for an error, since we show the quick-trace
+        for both event types
+        We occasionally have to get the nodestore data, so this function returns
+        the nodestore event as well so that we're doing that in one location.
         """
-        return {}
+        transaction_event = find_event(transactions, lambda item: item["id"] == event_id)
+        if transaction_event is not None:
+            return transaction_event, eventstore.get_event_by_id(
+                transaction_event["project.id"], transaction_event["id"]
+            )
+
+        # The event couldn't be found, it might be an error
+        error_event = find_event(errors, lambda item: item["id"] == event_id)
+        # Alright so we're looking at an error, time to see if we can find its transaction
+        if error_event:
+            # Unfortunately the only association from an event back to its transaction is name & span_id
+            # First maybe we got lucky and the error happened on the transaction's "span"
+            transaction_event = find_event(
+                transactions, lambda item: item["trace.span"] == error_event["trace.span"]
+            )
+            if transaction_event:
+                return transaction_event, eventstore.get_event_by_id(
+                    transaction_event["project.id"], transaction_event["id"]
+                )
+            # We didn't get lucky, time to talk to nodestore...
+            for transaction_event in transactions:
+                if transaction_event["transaction"] != error_event["transaction"]:
+                    continue
+
+                nodestore_event = eventstore.get_event_by_id(
+                    transaction_event["project.id"], transaction_event["id"]
+                )
+                for span in nodestore_event.data.get("spans", []):
+                    if span["span_id"] == error_event["trace.span"]:
+                        return transaction_event, nodestore_event
+
+        # The current event couldn't be found in errors or transactions
+        raise Http404()
+
+    def get_errors(self, organization, trace_id, params, current_event, event_id):
+        """Get errors for this trace
+
+        We try to optimize the light view's error retrieval, compared to
+        the full trace, so we try to get as few errors as possible
+        """
+        with sentry_sdk.start_span(op="discover", description="getting trace errors"):
+            with self.handle_query_errors():
+                # Search for errors for the current event if its a transaction, otherwise look for the exact error
```

Comment:
This won't get the errors on the child transactions like the full trace right? Not a huge deal but if this is true, then the loading states there will have the same problem where it'll appear without errors, and errors will pop in a little later.

Reply:
It won't, I'll backlog something to do the best effort association (ie. the error events trace.span == transaction.span) but adding errors is already pretty slow cause of the 2x snuba query and I try to keep the errors query light ([either txn name or id](https://github.com/getsentry/sentry/pull/24590/files#diff-619b86f549a9f11a566ca757be8da43a63bcaf6e4d7d3fe88a4222937a0209c6R214-R217)) so this best effort association isn't possible until I consolidate the snuba queries (also a backlogged task)